runtime/tests/checker/interface_test.go

```diff
@@ -2689,3 +2689,1069 @@ func TestCheckBadStructInterface(t *testing.T) {
+                        return 3
+                    }
+                }
+            }
+
+            contract interface B {
+                struct BNested {
+                    pub fun test(): Int {
+                        return 4
+                    }
+                }
+            }
+
+            contract interface C: A, B {}
+
+            contract X: C {
+                struct ANested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+
+                struct BNested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("type requirement multiple not conforming", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            contract interface A {
+                struct ANested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+
+            contract interface B {
+                struct BNested {
+                    pub fun test(): Int {
+                        return 4
+                    }
+                }
+            }
+
+            contract interface C: A, B {}
+
+            contract X: C {
+                struct ANested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+
+           contract Y: C {
+                struct BNested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+        `)
+
+		errs := RequireCheckerErrors(t, err, 2)
+		assert.IsType(t, &sema.ConformanceError{}, errs[0])
+		assert.IsType(t, &sema.ConformanceError{}, errs[1])
+	})
+
+	t.Run("nested struct conflicting", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            contract interface A {
+                struct Nested {
+                    pub fun test(): Int {
+                        return 3
+                    }
+                }
+            }
+
+            contract interface B: A {
+                struct Nested {
+                    pub fun test(): String {
+                        return "three"
+                    }
+                }
+            }
+        `)
+
+		errs := RequireCheckerErrors(t, err, 1)
+		memberConflictError := &sema.InterfaceMemberConflictError{}
+		require.ErrorAs(t, errs[0], &memberConflictError)
+		assert.Equal(t, common.DeclarationKindStructure, memberConflictError.MemberKind)
+		assert.Equal(t, common.DeclarationKindStructure, memberConflictError.ConflictingMemberKind)
+	})
+
+	t.Run("nested resource interface conflicting", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            contract interface A {
+                resource interface Nested {
```

Comment:
We should add tests for when a nested type interface inherits from another interface, particularly one nested inside a type that is also present inside the containing contract's type heirarchy. E.g.:

```
contract interface A {
    resource interface X: B.Y {}
}

contract interface B: A {
    resource interface Y {}
    resource interface X: Y {}
}
```

Should this count as a cyclical conformance?

Reply:
This is not cyclic, because `B.Y` is just a qualified name (a grouping), and neither `A` nor `A.X` directly depends on `B`.

Anyway, that example is a static error, because there's a conflict in type requirements: both `A` and `B` define their own `X` requirements. Can have only one of them.

Added a test with a simplified version of the example in c535fb201ed7837b5b9b65bc852a796b801fc8da.